src/migrations/1733808052981-item-tags-to-tag.ts

```diff
@@ -0,0 +1,134 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class Migrations1733808052981 implements MigrationInterface {
+  name = 'item-tags-to-tag-1733808052981';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // insert distinct tags
+    await queryRunner.query(`
+        insert into tag (name, category)
+        select DISTINCT(
+            trim(
+                -- replace faulty breaking lines into spaces
+                replace(
+                    unnest(
+                        -- break comma-separated tags into multiple tags 
+                        string_to_array(
+                            -- remove resulting quotes (") 
+                            substr(
+                                jsonb_array_elements(settings::jsonb->'tags')::text,
+                                2,
+                                length(jsonb_array_elements( settings::jsonb->'tags' )::text) - 2
+                            ),','
+                        )
+                    ),'\n',' '
+                )
+            )
+        ), 'discipline'::tag_category_enum from item
+
+        where settings::jsonb->'tags' is not null 
+        `);
+
+    // insert item_tag relation
+    await queryRunner.query(`
+        insert into item_tag (tag_id, item_id)
+        select  tag.id, iid from (
+        -- get all items and their corresponding tag entries
+        select item.id as iid,  trim(
+                replace(
+                    unnest(
+                        string_to_array(
+                            substr(
+                                jsonb_array_elements(settings::jsonb->'tags')::text,
+                                2,
+                                length(jsonb_array_elements( settings::jsonb->'tags' )::text) - 2
+                                ),','
+                            )
+                        ),'\n',' '
+                    )
+                ) as t
+            from item 
+        
+        where settings::jsonb->'tags' is not null 
+        
+        ) as tags
+        left join tag on tags.t = tag.name
+        `);
+
+    // manually migrate existing disciplines to level and resource type tags to populate
+    for (const [category, tags] of [
+      [
+        'level',
+        [
+          '2º BACH',
+          '3 ESO',
+          '4. Klasse',
+          '5º curso',
+          '7. Klasse',
+          'A1 ELE',
+          'B1',
+          'CDD course. B2',
+          'collège',
+          'ELE. A1.',
+          'formación profesional',
+          'Lehrplan 21',
+          'MA.2.A.1.i',
+          'MA.2.C.2.h',
+          'Sek1',
+          'Sekundarstufe 1',
```

Comment:
These are from the current levels in the database ?

Reply:
Yes, from the current tags.